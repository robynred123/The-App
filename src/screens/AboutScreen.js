import React from 'react';
import { Text, View, Image, StyleSheet, ScrollView, Linking, TouchableOpacity } from 'react-native';
import { FontAwesomeIcon } from '@fortawesome/react-native-fontawesome';
import { faStar } from '@fortawesome/free-solid-svg-icons';
import { interests, technologies } from '../data/data';

const AboutScreen = () => {
  return (
    <ScrollView>
      <View style={styles.container}>
        <Text style={styles.title}>About Me</Text>
        <Image source={require('../../assets/about.jpg')} style={styles.image} />
        <Text style={styles.text}>
          My name is Robyn, I am an apprentice software engineer at Asda working towards my level 4 qualification.
        </Text>
        <Text style={styles.text}>
          I am most comfortable working on react and react-native applications, but I do have experience in Node.js, Java, and C#.
        </Text>
      </View>

      <View style={{padding: 10}}>
        <Text style={styles.title}>Technologies and Skills I'm Interested In</Text>
        {technologies.map((technology) => (
           <Text style={styles.list}>{<FontAwesomeIcon icon={faStar} fill='#f1c232' />} { technology } </Text>
        ))}
      </View>

      <View style={{padding: 10}}>
        <Text style={styles.title}>My Interests</Text>
        {interests.map((interest) => (
           <Text style={styles.list}>{<FontAwesomeIcon icon={faStar} fill='#f1c232' />} { interest } </Text>
        ))}
      </View>

      <View style={styles.link}>
        <Text> 
          For More Information, Visit My </Text>
          <TouchableOpacity onPress={() => Linking.openURL('https://rbursnell.netlify.app/')}>
            <Text style={{...styles.text, color:'#b067d8'}}>
              Website
            </Text>
          </TouchableOpacity>
      </View>
    </ScrollView>
  )
}
  
const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
  },
  title: {
    fontSize: 20,
    fontWeight: '700',
    justifyContent: 'center',
    padding: 5,
    textAlign: 'center'
  },
  image: {
    width: 300,
    height: 200,
    margin: 10
  },
  text: {
    marginHorizontal: 10,
    padding: 5,
    fontSize: 15,
    flexWrap: 'wrap',
    textAlign: 'center'
  },
  list: {
    textAlign: 'left',
    alignItems: 'flex-start',
    marginLeft: 10
  },
  link: {
    alignItems: 'center',
    flexDirection: 'row',
  }
});

export default AboutScreen