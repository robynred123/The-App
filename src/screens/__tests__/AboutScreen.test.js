import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import AboutScreen from '../AboutScreen';
import { interests, technologies } from '';

Enzyme.configure({ adapter: new Adapter() })

jest.mock('@fortawesome/react-native-fontawesome', () => ({
    FontAwesomeIcon: ''
}))

jest.mock('@fortawesome/free-solid-svg-icons', () => ({
    faStar: 'StarIcon'
}))

jest.mock('../../data/data', () => ({
    interests: [
        'gardening',
        'cross-country running'
    ],
    technologies: [
        'advanced neural networks',
        'integration layer development'
    ]
}))

describe('<AboutScreen />', () => {
    const component = shallow(<AboutScreen onPress={onPress}/>)

    it('should render the AboutScreen without crashing', () => {
        expect(component).toMatchSnapshot()
    });

    it('should render the AboutScreen with a title of "About Me"', () => {
        expect(component.find('Text').first().children().text()).toEqual('About Me')
    });

    it('should render the AboutScreen with an image', () => {
        expect(component.find('Image')).toBeTruthy()
    });

    it('should render the AboutScreen with a list of technologies', () => {
        expect(component.find('.technology').first().dive().text()).toEqual('advanced neural networks')
        expect(component.find('.technology').at(1).dive().text()).toEqual('integration layer development')
    });

    it('should render the AboutScreen with a list of interests', () => {
        expect(component.find('.interest').first().dive().text()).toEqual('gardening')
        expect(component.find('.interest').at(1).dive().text()).toEqual('cross-country running')
    });

    it('should render the AboutScreen with a link to my website', () => {
        expect(component.find('.website').dive().text()).toEqual(' For More Information, Visit My ')
        expect(component.find('ForwardRef').children().dive().text()).toEqual('Website')
    });
})