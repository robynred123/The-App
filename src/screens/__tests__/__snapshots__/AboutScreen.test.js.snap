// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AboutScreen /> should render the AboutScreen without crashing 1`] = `
<ScrollViewMock>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 20,
          "fontWeight": "700",
          "justifyContent": "center",
          "padding": 5,
          "textAlign": "center",
        }
      }
    >
      About Me
    </Text>
    <Image
      source={1}
      style={
        Object {
          "height": 200,
          "margin": 10,
          "width": 300,
        }
      }
    />
    <Text
      style={
        Object {
          "flexWrap": "wrap",
          "fontSize": 15,
          "padding": 10,
          "textAlign": "center",
        }
      }
    >
      My name is Robyn, I am an apprentice software engineer at Asda working towards my level 4 qualification.
    </Text>
    <Text
      style={
        Object {
          "flexWrap": "wrap",
          "fontSize": 15,
          "padding": 10,
          "textAlign": "center",
        }
      }
    >
      I am most comfortable working on react and react-native applications, but I do have experience in Node.js, Java, and C#.
    </Text>
  </View>
  <View
    style={
      Object {
        "padding": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 20,
          "fontWeight": "700",
          "justifyContent": "center",
          "padding": 5,
          "textAlign": "center",
        }
      }
    >
      Technologies and Skills I'm Interested In
    </Text>
    <Text
      className="technology"
      key="0"
      style={
        Object {
          "alignItems": "flex-start",
          "marginLeft": 10,
          "textAlign": "left",
        }
      }
    >
      <Component
        fill="#f1c232"
        icon="StarIcon"
      />
      advanced neural networks
    </Text>
    <Text
      className="technology"
      key="1"
      style={
        Object {
          "alignItems": "flex-start",
          "marginLeft": 10,
          "textAlign": "left",
        }
      }
    >
      <Component
        fill="#f1c232"
        icon="StarIcon"
      />
      integration layer development
    </Text>
  </View>
  <View
    style={
      Object {
        "padding": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 20,
          "fontWeight": "700",
          "justifyContent": "center",
          "padding": 5,
          "textAlign": "center",
        }
      }
    >
      My Interests
    </Text>
    <Text
      className="interest"
      key="0"
      style={
        Object {
          "alignItems": "flex-start",
          "marginLeft": 10,
          "textAlign": "left",
        }
      }
    >
      <Component
        fill="#f1c232"
        icon="StarIcon"
      />
      gardening
    </Text>
    <Text
      className="interest"
      key="1"
      style={
        Object {
          "alignItems": "flex-start",
          "marginLeft": 10,
          "textAlign": "left",
        }
      }
    >
      <Component
        fill="#f1c232"
        icon="StarIcon"
      />
      cross-country running
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
          "paddingHorizontal": 10,
          "paddingVertical": 20,
        }
      }
    >
      <Text
        className="website"
        style={
          Object {
            "flexWrap": "wrap",
            "fontSize": 15,
            "textAlign": "center",
          }
        }
      >
         For More Information, Visit My 
      </Text>
      <ForwardRef
        onPress={[Function]}
      >
        <Text
          style={
            Object {
              "color": "#b067d8",
              "flexWrap": "wrap",
              "fontSize": 15,
              "textAlign": "center",
            }
          }
        >
          Website
        </Text>
      </ForwardRef>
    </View>
  </View>
</ScrollViewMock>
`;
